components/agendor/agendor.app.mjs

```diff
@@ -0,0 +1,243 @@
+          value: person.id,
+        }));
+      },
+    },
+    organizationId: {
+      type: "string",
+      label: "Organization ID",
+      description: "The ID of the organization to retrieve.",
+      async options(prevContext) {
+        const organizations = await this.listOrganizations(prevContext.page + 1);
+        return organizations.data.map((organization) => ({
+          label: organization.name,
+          value: organization.id,
+        }));
+      },
+    },
+    userId: {
+      type: "string",
+      label: "Owner User",
+      description: "User ID or email of the owner of this organization.",
+      optional: true,
+      async options() {
+        const users = await this.listUsers();
+        return users.data.map((user) => ({
+          label: user.name,
+          value: user.id,
+        }));
+      },
+    },
+    ranking: {
+      type: "string",
+      label: "Ranking",
+      description: "Ranking displayed as stars in the page.",
+      optional: true,
+      options: [
+        "0",
+        "1",
+        "2",
+        "3",
+        "4",
+        "5",
+      ],
+    },
+    description: {
+      type: "string",
+      label: "Description",
+      description: "Your description of this organization.",
+      optional: true,
+    },
+    contact: {
+      type: "object",
+      label: "Contact",
+      description: "Contact information.",
+      optional: true,
+    },
+    address: {
+      type: "object",
+      label: "Address",
+      description: "Address information.",
+      optional: true,
+    },
+    leadOrigin: {
+      type: "string",
+      label: "Lead Origin",
+      description: "Lead origin.",
+      optional: true,
+      async options() {
+        const leadOrigins = await this.listLeadOrigins();
+        return leadOrigins.data.map((leadOrigin) => ({
+          label: leadOrigin.name,
+          value: leadOrigin.id,
+        }));
+      },
+    },
+    category: {
+      type: "string",
+      label: "Category",
+      description: "Category ID or name.",
+      optional: true,
+      async options() {
+        const categories = await this.listCategories();
+        return categories.data.map((category) => ({
+          label: category.name,
+          value: category.id,
+        }));
+      },
+    },
+    sector: {
+      type: "string",
+      label: "Sector",
+      description: "Sector ID or name.",
+      optional: true,
+      async options() {
+        const sectors = await this.listSectors();
+        return sectors.data.map((sector) => ({
+          label: sector.name,
+          value: sector.id,
+        }));
+      },
+    },
+    product: {
+      type: "string",
+      label: "Product",
+      description: "Array of product IDs.",
+      optional: true,
+      async options(prevContext) {
+        const products = await this.listProducts(prevContext.page + 1);
+        return products.data.map((product) => ({
+          label: product.name,
+          value: product.id,
```

Comment:
```suggestion
        const { data } = await this.listProducts(prevContext.page + 1);
        return data.map(({
            name: label, id: value,
        }) => ({
          label,
          value,
```

Reply:
If you don't mind, I would like to keep it as it is now. I need to break more lines in order to make this destructor because of our lint rules. In the end, it is a little bit harder to read using destructors in this case.
But if you really have a strong opinion about it. I can change it.